include/tvm/node/container.h

```diff
@@ -49,11 +50,24 @@ using runtime::ObjectRef;
 using runtime::String;
 using runtime::StringObj;
 
+struct MapObjectHash {
```

Comment:
can we simply change ObjectHash to this impl?

Reply:
One thing i am not sure of: `ObjectHash` and `ObjectEqual` are defined in object.h, do we really want to make it depend on classes defined in `container.h`?

Reply:
A workaround can be checking if `type_index_` is `kRuntimeString`

Reply:
Per our offline discussion, we decided that we use a new name `ObjectPtrEqual`/`ObjectPtrHash` for previous `ObjectEqual`/`ObjectHash`, and the new `MapObjectHash`/`MapObjectEqual` is renamed as `ObjectEqual`/`ObjectHash`.